packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -34,6 +35,359 @@ enum DayPeriod {
   pm,
 }
 
+/// The list of known time of day formats.
+///
+/// This time picker has configurations optimized for these formats. Any other
+/// formats will default to [h_colon_mm_space_a].
+abstract class TimeOfDayFormats {
+  /// The default en_US format.
+  ///
+  /// Used when the configuration for the currently specified
+  /// [MaterialLocalizations.timeOfDayFormat] is not available.
+  static const String h_colon_mm_space_a = 'h:mm a';
+
+  // Reusable patterns.
+  static const String HH_colon_mm = 'HH:mm';
+  static const String H_colon_mm = 'H:mm';
+  static const String HH_dot_mm = 'HH.mm';
+  static const String a_space_h_colon_mm = 'a h:mm';
+
+  // Locale-specific.
+  static const String frenchCanadian = "HH 'h' mm";
+  static const String thai = 'HH:mm น.';
+
+  /// Enumerates formats supported by the material time picker.
+  static const List<String> supportedFormats = const <String>[
+    h_colon_mm_space_a,
+    HH_colon_mm,
+    H_colon_mm,
+    HH_dot_mm,
+    a_space_h_colon_mm,
+    frenchCanadian,
+  ];
+}
+
+class _TimePickerHeaderFormat {
+  const _TimePickerHeaderFormat(this.centrepieceIndex, this.pieces);
+
+  /// Pieces that constitute a time picker header.
+  final List<_TimePickerHeaderPiece> pieces;
+
+  /// Index into the [pieces] list pointing at the piece that contains the
+  /// pivot fragment.
+  final int centrepieceIndex;
```

Comment:
I'd put the members in the same order as they are in the constructor's argument list.